TLM/TLM/Patcher.cs

```diff
@@ -0,0 +1,93 @@
+namespace TrafficManager {
+    using ColossalFramework;
+    using ColossalFramework.UI;
+    using CSUtil.Commons;
+    using HarmonyLib;
+    using System;
+    using System.Collections.Generic;
+    using System.Reflection;
+    using TrafficManager.RedirectionFramework;
+    using TrafficManager.Util;
+
+    public class Patcher {
+        public static Patcher Instance { get; private set; }
+
+        public static Patcher Create() => Instance = new Patcher();
+
+        private const string HARMONY_ID = "de.viathinksoft.tmpe";
+
+        private bool initialized_ = false;
+
+        public void Install() {
+            if (initialized_) {
+                return;
+            }
+
+            Log.Info("Init detours");
+            bool fail = false;
+
+            try {
+#if DEBUG
+                Harmony.DEBUG = true;
+#endif
+                // Harmony attribute-driven patching
+                Log.Info($"Performing Harmony attribute-driven patching");
+                var harmony = new Harmony(HARMONY_ID);
+                Shortcuts.Assert(harmony != null, "HarmonyInst!=null");
+                harmony.PatchAll();
+                Log.Info($"Harmony attribute-driven patching successfull!");
+            }
+            catch (Exception e) {
+                Log.Error("Could not deploy Harmony patches");
+                Log.Info(e.Message);
+                Log.Info(e.StackTrace);
+                fail = true;
+                throw e;
```

Comment:
How are we going to process `fail` (line 59 below) if we throw an exceptionhere? (Alternatively, why are we bothing to set `fail = true` on line 44?)

Reply:
Ooops. forgot to delete debug code!

Reply:
I'd rather the much simpler approach of
```
new Harmony(HARMONY_ID).PatchAll();
AssemblyRedirector.Deploy();
```
I was just trying not to change existing code.

Reply:
@aubergine10 I removed the line you requested.